### PR TITLE
Add support for native DNS.

### PR DESCRIPTION
**Pull request by shettyg**
OVN can intercept DNS packets and respond to them.
This patch starts with simply responding to $SERVICE_NAME.NAMESPACE

### CODE REVIEW

**Comment by shettyg** [comment]
I did a round of testing and looks to work fine. The question is, whether we need this feature though. Or should we make it optional.

**Comment by rajatchopra** [comment]
It's a useful feature but it should be optional I believe. Kubernetes deployments often rig up their own dns systems so we should not interfere in that case.
Will try this out.

File: go-controller/pkg/ovn/service.go [hunk 1]
```diff
@@ -1,11 +1,88 @@
...
+	dnsRecord = strings.TrimSpace(string(uuid))
+
+	return dnsRecord
+}
+
+func setDNSRecord(name, namespace, IP string) {
+	if name == "" || IP == "" || namespace == "" {
+		return
+	}
+	uuid := getDNSTable()
```
Comment (rajatchopra) [inline go-controller/pkg/ovn/service.go]: We should cache this.

Comment (shettyg) [reply go-controller/pkg/ovn/service.go]: I think it is cached.

**Comment by shettyg** [comment]
>Will try this out.
FYI,  containers need clear connectivity to their local DNS (in /etc/resolv.conf) for this to work.

File: go-controller/pkg/ovn/service.go (hunk 1 above)
Comment (rajatchopra) [reply go-controller/pkg/ovn/service.go]: Yes. I see it. Wrong call.

**Comment by rajatchopra** [comment]
> FYI, containers need clear connectivity to their local DNS (in /etc/resolv.conf) for this to work.

Can you elaborate on this? What should the nameserver be?
The dns records exist just fine in the table.

**Comment by shettyg** [comment]
Closing this. Will revisit after simplifying everything else.

**Pull request closed without merging**